devicetypes/smartenit/iot8-z.src/iot8-z.groovy

```diff
@@ -0,0 +1,223 @@
+			}
+			if (childDevice) {
+				if (eventDescMap?.clusterInt == BINARY_INPUT_CLUSTER) {
+					if (eventDescMap?.value == "00") {
+						def map = createEvent(name: "contact", value: "open")
+						childDevice.sendEvent(map)
+					} else if (eventDescMap?.value == "01") {
+						def map = createEvent(name: "contact", value: "closed")
+						childDevice.sendEvent(map)
+					}
+				} else if (eventDescMap?.clusterInt == ANALOG_INPUT_CLUSTER) {
+					long convertedValue = Long.parseLong(eventDescMap?.value, 16)
+					Float percentage = Float.intBitsToFloat(convertedValue.intValue())
+					percentage = (percentage / 1.60) * 100.0
+					def ceilingVal = Math.ceil(percentage)
+					if (ceilingVal > 100.0) {
+						ceilingVal = 100.0
+					}
+
+					int intValue = (int) ceilingVal
+
+					def map = createEvent(name: "inputValue", value: intValue)
+					childDevice.sendEvent(map)
+				}
+			}
+		}
+	}
+}
+
+def on() {
+	zigbee.on()
+}
+
+def off() {
+	zigbee.off()
+}
+
+def childOn(String dni) {
+	def childEndpoint = getChildEndpoint(dni)
+	zigbee.command(zigbee.ONOFF_CLUSTER, 0x01, "", [destEndpoint: childEndpoint])
+}
+
+def childOff(String dni) {
+	def childEndpoint = getChildEndpoint(dni)
+	zigbee.command(zigbee.ONOFF_CLUSTER, 0x00, "", [destEndpoint: childEndpoint])
+}
+
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	def refreshCommands = zigbee.onOffRefresh() 
+	def numberOfChildDevices = 8
+
+	for (def endpoint : 2..numberOfChildDevices) {
+		refreshCommands += zigbee.readAttribute(zigbee.ONOFF_CLUSTER, ONOFF_ATTRIBUTE, [destEndpoint: endpoint])
+	}
+	for (def endpoint : 1..4) {
+		refreshCommands += zigbee.readAttribute(BINARY_INPUT_CLUSTER, PRESENT_VALUE_ATTRIBUTE, [destEndpoint: endpoint])
+	}
+
+	refreshCommands += zigbee.readAttribute(ANALOG_INPUT_CLUSTER, PRESENT_VALUE_ATTRIBUTE, [destEndpoint: 0x0001]);
+	refreshCommands += zigbee.readAttribute(ANALOG_INPUT_CLUSTER, PRESENT_VALUE_ATTRIBUTE, [destEndpoint: 0x0002]);
+	log.debug "refreshCommands: $refreshCommands"
+
+	return refreshCommands
+}
+
+private void createChildDevices() {
+	def numberOfChildDevices = 8
+
+	for (def endpoint: 2..numberOfChildDevices) {
+		try {
+			if (endpoint == 2) {
+				addChildDevice("Smartenit", "IOT8-Z-child-analog-contact-switch", "${device.deviceNetworkId}:0${endpoint}", device.hubId,
+					[completedSetup: true,
+					label: "${device.displayName} ${endpoint}",
+					isComponent: false
+					])
+			} else if (endpoint >= 3 && endpoint <= 4) {
+				addChildDevice("Smartenit", "IOT8-Z-child-contact-switch", "${device.deviceNetworkId}:0${endpoint}", device.hubId,
+					[completedSetup: true,
+					label: "${device.displayName} ${endpoint}",
+					isComponent: false
+					])
+			} else if (endpoint >= 5 && endpoint <= 8) {
+				addChildDevice("smartthings", "ZigBee Switch", "${device.deviceNetworkId}:0${endpoint}", device.hubId,
```

Comment:
I wouldn't expect this to work well. If the user tries to turn these switches on and off, I expect the command will fail. Perhaps you meant to use `"Child Switch"`?

Reply:
Oh you're right. This was leftover from my testing and it's been working for me so I completely missed it. Let me make the modification real quick.